client/src/pages/Bookings/Bookings.tsx

```diff
@@ -0,0 +1,69 @@
+import { useState, useEffect } from 'react';
+import { getRequests } from '../../helpers/APICalls/request';
+import Container from '@material-ui/core/Container';
+import Grid from '@material-ui/core/Grid';
+import Box from '@material-ui/core/Box';
+import useStyles from './useStyles';
+import React from 'react';
+import Calendar from './Calendar';
+import NextBookings from './NextBookings';
+import ManageBookings from './ManageBookings';
+import { RequestApiDataSuccess } from '../../interface/Requests';
+
+export default function Bookings(): JSX.Element {
+  const classes = useStyles();
+  const resModel: RequestApiDataSuccess = {
+    nextBooking: {},
+    currentBookings: [],
+    pastBookings: [],
+  };
+  const [requests, setRequest] = useState<RequestApiDataSuccess>(resModel);
+  const returnRequestType = (success): RequestApiDataSuccess => success.resModel;
+
+  const updateStatusState = (id: string, status: string, sectionName: string) => {
+    const requestCopy = { ...requests };
+    if (id == requestCopy.nextBooking?._id) {
+      requestCopy.nextBooking.status = status;
+      setRequest(requestCopy);
+      return;
+    }
+
+    const index = requestCopy[sectionName].findIndex((booking) => booking._id === id);
+    if (index >= 0) {
+      requestCopy[sectionName][index].status = status;
+      setRequest(requestCopy);
+      return;
+    }
+  };
+
+  useEffect(() => {
+    const fetchRequests = async () => {
+      const success = await getRequests();
+      const resModelSuccess = returnRequestType(success);
```

Comment:
Wouldn't restructuring the response data here make it a lot simpler as opposed to passing it into a function?